### DISCUSSION TITLE
On/off when run_this_before_entering_active_state

### DISCUSSION THREAD

**Discussion by ghost** [mikebrady/shairport-sync#1637] [Show and tell]
I want to show how you can use your raspberry to trigger your amp to turn on when you run `run_this_before_entering_active_state` and turn it off when you run `run_this_after_exiting_active_state`

![IMG_0016](https://user-images.githubusercontent.com/55313117/220666821-7139bae4-5915-4052-8597-c4eef0e13a02.jpeg)

My system is:

Purify EVAL1
smps1200a400
SMSL SU-8s
Raspberry PI4
Elac FS407

The SMPS1200a400 has the ability to turn on at 0V and go into standby at 3V. And that is the starting point for this "guide". If you need to switch on at 12V and go into standby at 0V, that is also possible, but that is beyond the scope of this guide. (Suggestions and examples are welcome)

I use GPIO/BCM pin 21/GND39 on my Raspberry which is connected to GPIO/GND PIN J5.1/J5.5 on my smps1200a400 using two of these [ https://www.conrad.com/p/tru -components-718672-35-mm-audio-jack-socket-vertical-vertical-number-of-pins-num-2-mono-black-1-pcs-1567107](url)

`sudo usermod -a -G gpio shairport-sync`

**On via GPIO**

`sudo mkdir script`

`sudo nano script/amp_on.py`

```
#!/bin/python3
from gpiozero import OutputDevice
import sys

sys.path.append("/home/pi/script/amp_off.py")
AMP_TRIGGER_PIN = 21

amp = OutputDevice(AMP_TRIGGER_PIN, active_high=False, initial_value=False)

amp.on()
sys.exit()
```

**Off via GPIO**

`sudo nano script/amp_off.py`

```
#!/bin/python3
from gpiozero import OutputDevice
from signal import pause
import sys

sys.path.append("/home/pi/script/amp_on.py")

AMP_TRIGGER_PIN = 21

amp = OutputDevice(AMP_TRIGGER_PIN, active_high=False, initial_value=False)

amp.off()
pause()
```

```
sudo chmod +x script/amp_on.py
sudo chmod +x script/amp_off.py
```

**Off on boot via GPIO** 

`sudo mkdir logs`

`sudo nano script/off_on_boot.sh`

```
#!/bin/sh
# off_on_boot.sh
# navigate to home directory, then to this directory, then execute python script, then back home

cd /
cd home/pi/script
python3 amp_off.py
cd /
```

`sudo chmod 755 script/off_on_boot.sh`

`sudo crontab -e`

`@reboot sh /home/pi/script/off_on_boot.sh >/home/pi/logs/cronlog 2>&1`

`sudo nano /etc/shairport-sync.conf`

```
sessioncontrol =
{
        run_this_before_entering_active_state = “home/pi/script/amp_on.py";
        run_this_after_exiting_active_state = “home/pi/script/amp_off.py";
        active_state_timeout = 360.0;
};
```

**Comment by mikebrady** [comment]
Thanks for this!
